app/src/auth/components/ResetPasswordPage/index.jsx

```diff
@@ -15,150 +20,154 @@ import AuthLayout from '../AuthLayout'
+    token: '',
+}
+
+const ResetPasswordPage = ({ location: { search, pathname }, history: { replace }, redirectToLoginPage }: Props) => {
+    const {
+        errors,
+        form,
+        isProcessing,
+        setFieldError,
+        setFormField,
+        step,
+    } = useContext(Context)
+
+    const onFailure = useCallback(({ message }: Error) => {
+        setFieldError('confirmPassword', message)
+    }, [setFieldError])
 
-        setFormField('token', token, () => {
-            yup
-                .object()
-                .shape({
-                    token: yup.reach(schemas[0], 'token'),
-                })
-                .validate(this.props.form)
-                .then(
-                    () => {
-                        // To make sure that the resetPassword token doesn't stick in the browser history
-                        props.history.replace(props.location.pathname)
-                    },
-                    (error: yup.ValidationError) => {
-                        setFieldError('password', error.message)
-                    },
-                )
-        })
-    }
-
-    onFailure = (error: Error) => {
-        const { setFieldError } = this.props
-        setFieldError('confirmPassword', error.message)
-    }
-
-    submit = () => {
-        const { password, confirmPassword: password2, token: t } = this.props.form
+    const submit = useCallback(() => {
+        const { password, confirmPassword: password2, token: t } = form
 
         return post(routes.externalResetPassword(), {
             password,
             password2,
             t,
         }, false, true)
-    }
-
-    render() {
-        const {
-            setIsProcessing,
-            isProcessing,
-            step,
-            form,
-            errors,
-            setFieldError,
-            next,
-            prev,
-            setFormField,
-            // TODO: Sign the user in on success. Missing part: current reset password endpoint
-            //       doesn't respond with user instance so we don't have access to the email address.
-            redirectToLoginPage,
-        } = this.props
-
-        return (
-            <AuthLayout>
-                <AuthPanel
-                    currentStep={step}
-                    form={form}
-                    onPrev={prev}
-                    onNext={next}
-                    setIsProcessing={setIsProcessing}
-                    isProcessing={isProcessing}
-                    validationSchemas={schemas}
-                    onValidationError={setFieldError}
+    }, [form])
+
+    useEffect(() => {
+        const token = qs.parse(search).t || ''
+
+        // Set and validate token on mount.
+        setFormField('token', token)
+
+        yup
+            .object()
+            .shape({
+                token: yup.reach(schemas[0], 'token'),
+            })
+            .validate({
+                token,
+            })
+            .then(
+                () => {
```

Comment:
Should this check for unmount before doing anything after the async step?

Reply:
I guess! Fixed it as well as a couple of other instances of the same.. issue.